packages/assistify-ai/server/lib/SmartiAdapter.js

```diff
@@ -34,22 +34,93 @@ export class SmartiAdapter {
 	 */
 	static onMessage(message) {
 
-		//TODO is this always a new one, what about update
-		const helpRequest = RocketChat.models.HelpRequests.findOneByRoomId(message.rid);
-		const supportArea = helpRequest ? helpRequest.supportArea : undefined;
-		const requestBody = {
-			// TODO: Should this really be in the responsibility of the Adapter?
-			webhook_url: SmartiAdapter.rocketWebhookUrl,
-			message_id: message._id,
-			channel_id: message.rid,
-			user_id: message.u._id,
-			// username: message.u.username,
-			text: message.msg,
-			timestamp: message.ts,
-			origin: message.origin,
-			support_area: supportArea
+		//TODO trigger on message update, if needed
+		const requestBodyMessage = {
+			'id': message._id,
+			'time': message.ts,
+			'origin': 'User', //user.type,
+			'content': message.msg,
+			'user': {
+				'id': message.u._id
+			}
+			//,"private" : false
 		};
-		return SmartiProxy.propagateToSmarti(verbs.post, `rocket/${ SmartiAdapter.smartiKnowledgeDomain }`, requestBody);
+
+		SystemLogger.debug('Message:', requestBodyMessage);
+
+		const m = RocketChat.models.LivechatExternalMessage.findOneById(message.rid);
+		let analysisResult;
+		let conversationId;
+
+		// conversation exists for channel?
+		if (m && m.conversationId) {
+			conversationId = m.conversationId;
+		} else {
+			SystemLogger.debug('Smarti - Trying legacy service to retrieve conversation ID...');
+			const conversation = SmartiProxy.propagateToSmarti(verbs.get, `legacy/rocket.chat?channel_id=${ message.rid }`);
+			if (conversation && conversation.id) {
+				conversationId = conversation.id;
+			}
+		}
+
+		if (conversationId) {
+			SystemLogger.info('Conversation found for channel');
+			// add message to conversation
+			SmartiProxy.propagateToSmarti(verbs.post, `conversation/${ conversationId }/message`, requestBodyMessage);
```

Comment:
Now, only the message which is triggering the hook is being transmitted. Earlier, always all messages of the conversation have been submitted.
Though the new approach is technically fine, it raises a new question: What happens if Smarti was not able to process a previous message (e. g. since it was unavailable or not able to process the request)? I believe all messages newer than the latest message recorded for the conversation should be transmitted.